### PR TITLE
Feature/user-exist

### PR DESCRIPTION
**Pull request by misspelz**
Email Check Endpoint Integration and Modal

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -22,6 +23,7 @@ function App() {
   return isOnline ? (
     <>
       {scrollOnRoute}
+      <Toaster position="top-center" reverseOrder={false} />
```
Comment (edustipend-root) [inline src/App.js:26]: What is this doing?

File: src/components/ApplicationSteps/Step4Application/Internals/constants.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ export const constants = {
   GENDER_TEXT: `We want to be able to support everyone fairly regardless of gender.`,
   TWITTER_TEXT: `If you're not on Twitter, please type N/A. If you're on some other social media platform, please share and indicate`,
   MONTHS_OPTION: ['JAN', 'FEB', 'MAR', 'APR', 'MAY', 'JUN', 'JUL', 'AUG', 'SEP', 'OCT', 'NOV', 'DEC'],
-  GENDER_OPTION: ['Male', 'Female', 'Others'],
+  GENDER_OPTION: ['Male', 'Female'],
```
Comment (edustipend-root) [inline src/components/ApplicationSteps/Step4Application/Internals/constants.js:31]: Gender options are `Male`, `Female` and `Non-Binary`

File: src/utils/checkTokenExp/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const checkTokenExp = (tokenTime) => {
+  const today = new Date();
+  const tokenExpDate = new Date(tokenTime * 1000);
+  console.log(today, tokenExpDate);
+  if (today >= tokenExpDate) {
```
Comment (edustipend-root) [inline src/utils/checkTokenExp/index.js:5]: What's the use of this `if` condition and also can we remove the console logs.

File: src/utils/getStateIdentifier/index.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  { label: 'Oyo', value: 'OY' },
+  { label: 'Plateau', value: 'PL' },
+  { label: 'Rivers', value: 'RI' },
+  { label: 'Sokoto', value: 'SO' },
+  { label: 'Taraba', value: 'TA' },
+  { label: 'Yobe', value: 'YO' },
+  { label: 'Zamfara', value: 'ZA' }
+];
+
+export const getStateIdentifier = (stateLabel) => {
```
Comment (edustipend-root) [inline src/utils/getStateIdentifier/index.js:40]: Please add a test for this utility @misspelz.

**Review by edustipend-root — CHANGES_REQUESTED** [review]

File: src/sections/EmailExist/constants.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export const TestId = {
```
Comment (edustipend-root) [inline src/sections/EmailExist/constants.js:1]: Why is this named `TestId`?

File: src/sections/EmailExist/constants.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+  EMAIL_EXIST_TEXT: 'Your Email Exist!',
```
Comment (edustipend-root) [inline src/sections/EmailExist/constants.js:2]: Please change this text from `Your Email Exist!` to `Hey! We know you`

File: src/sections/EmailExist/constants.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+  LOGIN_TEXT: 'Kindly, Log In'
```
Comment (edustipend-root) [inline src/sections/EmailExist/constants.js:3]: Please change this text from `Kindly, Log In` to `Click on the button below to login to your dashboard to view your current application(s)`

**Comment by edustipend-root** [comment]
@misspelz, thanks for this PR. There are a lot of changes in this PR that don't seem related to the work for checking if a user exists. 

That said, I've added some comments - please address those so we can get this merged in.

Also, you add some extra top margin above the `Login` button in the screenshot you attached?

File: src/components/ApplicationSteps/Step4Application/Step4Application.jsx [hunk 8]
```diff
@@ -78,6 +78,8 @@ export const Step4Application = () => {
     howDidYouHear
   };
 
+  console.log("userDetails", userDetails)
```
[the remark this answers is not in the document]
Comment (misspelz) [reply src/components/ApplicationSteps/Step4Application/Step4Application.jsx:81]: Done

File: src/components/ApplicationSteps/Step5Application/Internals/EmailVerified/EmailVerified.jsx [hunk 9]
```diff
@@ -23,23 +23,23 @@ export const EmailVerified = () => {
...
-      });
-
+      const res = await postData(
+        `user/verify?jwt${emailToken}`,
+        {
+          username: email
+        },
+        false
+      );
+      console.log(res);
```
[the remark this answers is not in the document]
Comment (misspelz) [reply src/components/ApplicationSteps/Step5Application/Internals/EmailVerified/EmailVerified.jsx:39]: Done

File: src/App.js (hunk 1 above)
Comment (misspelz) [reply src/App.js:26]: Ezeh added it

File: src/components/ApplicationSteps/Step4Application/Internals/constants.js (hunk 2 above)
Comment (misspelz) [reply src/components/ApplicationSteps/Step4Application/Internals/constants.js:31]: Done

File: src/sections/EmailExist/constants.js (hunk 5 above)
Comment (misspelz) [reply src/sections/EmailExist/constants.js:1]: Renamed

File: src/sections/EmailExist/constants.js (hunk 7 above)
Comment (misspelz) [reply src/sections/EmailExist/constants.js:3]: Done

File: src/utils/checkTokenExp/index.js (hunk 3 above)
Comment (misspelz) [reply src/utils/checkTokenExp/index.js:5]: File has been deleted

File: src/sections/EmailExist/constants.js (hunk 6 above)
Comment (misspelz) [reply src/sections/EmailExist/constants.js:2]: Done

File: src/utils/getStateIdentifier/index.js (hunk 4 above)
Comment (misspelz) [reply src/utils/getStateIdentifier/index.js:40]: Done

**Comment by misspelz** [comment]
> @misspelz, thanks for this PR. There are a lot of changes in this PR that don't seem related to the work for checking if a user exists.
> 
> That said, I've added some comments - please address those so we can get this merged in.
> 
> Also, you add some extra top margin above the `Login` button in the screenshot you attached?

I actually pulled from Isiaq's repo so we can be in sync. That's why there are unrelated changes but we are working together to resolve all changes

**Pull request merged**